src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -506,7 +506,8 @@ internal override Expression VisitMemberAccess(MemberExpression m)
             }
             else
             {
-                this.builder.Append(ClientTypeUtil.GetServerDefinedName(m.Member));
+                var member = m.Expression.Type.GetMember(m.Member.Name).First();

```

Comment:
Can you confirm (ideally in tests) that this fix also works with interfaces, abstract classes, different levels of inheritance, etc.

Reply:
Sure I will